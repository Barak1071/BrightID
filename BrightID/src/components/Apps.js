// @flow

import * as React from 'react';
import { StyleSheet, View, Alert } from 'react-native';
import { connect } from 'react-redux';

import server from '../Api/server';
import api from '../Api/BrightId';

class Apps extends React.Component<Props> {
  static navigationOptions = () => ({
    title: 'Apps',
    headerRight: <View />,
  });

  async componentDidMount() {
    if (this.props.navigation.state.params) { // if 'params' is defined, the user came through a deep link
      const { host, context, id } = this.props.navigation.state.params;
      const oldHost = server.baseUrl;
      let contextInfo;
      try {
        server.update(host);
        contextInfo = await api.getContext(context);
      } catch (e) {
        console.log(e);
      } finally {
        server.update(oldHost);
      }

      if (contextInfo && contextInfo.verification) {
        Alert.alert(
          'App Verification?',
          `Do you want to allow ${context} to link the account with id ${id} to your BrightID verification?`,
          [
            {
              text: 'Yes',
              onPress: () => this.linkVerification(host, context, contextInfo, id),
            },
            {
              text: 'No',
              style: 'cancel',
              onPress: () => {
                this.props.navigation.goBack();
              },
            },
          ],
        );
      } else {
        this.props.navigation.goBack();
      }
    }
  }

  render() {
    return (
      <View style={styles.container} />
    );
  }

  async linkVerification(host, context, contextInfo, id) {
    const oldHost = server.baseUrl;
    try {
      server.update(host);
      const verification = await api.getVerification(context, id);
      const response = await fetch(`${contextInfo.verificationUrl}/${id}`, {
        method: 'PUT',
        body: JSON.stringify(verification),
        headers: {
          'Content-Type': 'application/json',
        },
      });
      if (!response.ok) {
        throw new Error(response.statusText);
      }
    } catch (e) {
      Alert.alert(
        `App verification failed`,
        `${e.message}\n${e.stack || ''}`,
        [
          {
            text: 'Dismiss',
            style: 'cancel',
            onPress: () => {
              this.props.navigation.goBack();
            },
          },
        ],
      );
    } finally {
      server.update(oldHost);
    }
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
  },
});

export default connect((state) => state.main)(Apps);
