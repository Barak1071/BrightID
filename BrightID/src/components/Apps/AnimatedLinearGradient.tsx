import React from 'react';
import { StyleSheet, Animated } from 'react-native';
// import LinearGradient from 'react-native-linear-gradient';

type AnimatedLinearGradientProps = {
  children?: React.ReactNode;
  containerStyle?: any;
  style?: any;
  colors: string[];
};

export default function AnimatedLinearGradient(
  props: AnimatedLinearGradientProps,
) {
  const noGradient = (
    <Animated.View style={[styles.simple, props.containerStyle]}>
      <Animated.View
        style={StyleSheet.flatten([styles.linearGradient, props.style])}
      >
        {props.children}
      </Animated.View>
    </Animated.View>
  );

  /*
  const gradient = (
    <Animated.View
      style={StyleSheet.flatten([styles.container, props.containerStyle])}
    >
      <LinearGradient
        colors={props.colors}
        style={StyleSheet.flatten([styles.linearGradient, props.style])}
      >
        {props.children}
      </LinearGradient>
    </Animated.View>
  );

   */

  return noGradient;
}

const styles = StyleSheet.create({
  simple: {
    backgroundColor: '#3E4481',
  },
  container: {
    flex: 1,
  },
  linearGradient: {
    flex: 1,
  },
});
