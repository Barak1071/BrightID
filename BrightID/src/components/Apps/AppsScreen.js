// @flow

import * as React from 'react';
import {
  Alert,
  Linking,
  StyleSheet,
  View,
  FlatList,
  Text,
  StatusBar,
} from 'react-native';
import { connect } from 'react-redux';
import EmptyList from '@/components/Helpers/EmptyList';
import Spinner from 'react-native-spinkit';
import { ORANGE, DEVICE_LARGE } from '@/utils/constants';
import { find, propEq } from 'ramda';
import AppCard from './AppCard';
import { handleAppContext } from './model';

export class AppsScreen extends React.Component<Prop, State> {
  componentDidMount() {
    const { navigation } = this.props;
    navigation.addListener('focus', async () => {
      this.handleDeepLink();
      Linking.addEventListener('url', this.handleDeepLink);
    });
    navigation.addListener('blur', async () => {
      Linking.removeEventListener('url', this.handleDeepLink);
    });
  }

  isValidContext = (context) => {
    const { apps } = this.props;
    const app = find(propEq('context', context))(apps);
    return app !== undefined;
  };

  handleDeepLink = () => {
    const { route, navigation } = this.props;
    console.log('params', route.params);
    if (route.params?.context) {
      const { context } = route.params;
      if (this.isValidContext(context)) {
        handleAppContext(route.params);
      } else {
        Alert.alert('Failed', `${context} is not a valid context!`);
      }
    }
    // reset params
    navigation.setParams({
      baseUrl: '',
      context: '',
      contextId: '',
    });
  };

  statusBar = () => {
    const { isSponsored, linkedApps } = this.props;
    const pendingLink = find(propEq('state', 'pending'))(linkedApps);
    let msg, waiting;
    if (pendingLink) {
      msg = `Linking your account in ${pendingLink.context}\n to your BrightID ...`;
      waiting = true;
    } else if (!isSponsored) {
      msg = "You're not sponsored.\nPlease find an app below to sponsor you.";
      waiting = false;
    } else {
      msg = '';
      waiting = false;
    }
    return msg.length > 0 ? (
      <View style={styles.statusContainer}>
        <Text style={styles.statusMessage}>{msg}</Text>
        <Spinner isVisible={waiting} size={48} type="Wave" color="#4a90e2" />
      </View>
    ) : (
      <View />
    );
  };

  isLinked = (app) => {
    const { linkedApps } = this.props;
    let linked = false;
    linkedApps.forEach((link) => {
      if (link.state === 'applied' && link.context === app.context) {
        linked = true;
      }
    });
    return linked;
  };

  render() {
    const { apps } = this.props;
    return (
      <>
        <StatusBar
          barStyle="light-content"
          backgroundColor={ORANGE}
          animated={true}
        />
        <View style={styles.orangeTop} />
        <View style={styles.container} testID="appsScreen">
          <this.statusBar />
          <FlatList
            data={apps}
            contentContainerStyle={{ paddingBottom: 50, flexGrow: 1 }}
            keyExtractor={({ name }, index) => name + index}
            showsHorizontalScrollIndicator={false}
            showsVerticalScrollIndicator={false}
            renderItem={({ item }) => (
              <AppCard {...item} isLinked={this.isLinked(item)} />
            )}
            ListEmptyComponent={<EmptyList title="No Apps" iconType="flask" />}
          />
        </View>
      </>
    );
  }
}

const styles = StyleSheet.create({
  orangeTop: {
    backgroundColor: ORANGE,
    height: DEVICE_LARGE ? 70 : 65,
    width: '100%',
    zIndex: 1,
  },
  container: {
    flex: 1,
    backgroundColor: '#fdfdfd',
    borderTopLeftRadius: 58,
    marginTop: -58,
    zIndex: 10,
    overflow: 'hidden',
  },
  centerItem: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  statusContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 20,
  },
  statusMessage: {
    fontFamily: 'ApexNew-Medium',
    textAlign: 'center',
    fontSize: 18,
    color: '#4a90e2',
  },
  linkingContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: 20,
  },
});

export default connect(({ apps }) => ({ ...apps }))(AppsScreen);
