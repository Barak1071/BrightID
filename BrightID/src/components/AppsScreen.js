// @flow

import * as React from 'react';
import { StyleSheet, View, Alert } from 'react-native';
import { connect } from 'react-redux';

import api from '../Api/BrightId';

class AppsScreen extends React.Component<Props> {
  static navigationOptions = () => ({
    title: 'Apps',
    headerRight: <View />,
  });

  async componentDidMount() {
    if (this.props.navigation.state.params) { // if 'params' is defined, the user came through a deep link
      const { baseUrl, context, id } = this.props.navigation.state.params;
      const oldBaseUrl = api.baseUrl;
      let contextInfo;
      try {
        api.baseUrl = baseUrl;
        contextInfo = await api.getContext(context);
      } catch (e) {
        console.log(e);
      } finally {
        api.baseUrl = oldBaseUrl;
      }

      if (contextInfo && contextInfo.verification) {
        Alert.alert(
          'App Verification?',
          `Do you want to allow ${context} to link the account with id ${id} to your BrightID verification?`,
          [
            {
              text: 'Yes',
              onPress: () => this.linkVerification(baseUrl, context, contextInfo, id),
            },
            {
              text: 'No',
              style: 'cancel',
              onPress: () => {
                this.props.navigation.goBack();
              },
            },
          ],
        );
      } else {
        this.props.navigation.goBack();
      }
    }
  }

  render() {
    return (
      <View style={styles.container} />
    );
  }

  async linkVerification(baseUrl, context, contextInfo, id) {
    const oldBaseUrl = api.baseUrl;
    if(contextInfo.isApp){
      // TODO: store the app's image locally,
      //  save the app in redux, so it can show in the app list
      //  until the user removes it.
    }
    try {
      api.baseUrl = baseUrl;
      const verification = await api.getVerification(context, id);
      const response = await fetch(`${contextInfo.verificationUrl}/${id}`, {
        method: 'PUT',
        body: JSON.stringify(verification),
        headers: {
          'Content-Type': 'application/json',
        },
      });
      if (!response.ok) {
        throw new Error(response.statusText);
      }
    } catch (e) {
      Alert.alert(
        `App verification failed`,
        `${e.message}\n${e.stack || ''}`,
        [
          {
            text: 'Dismiss',
            style: 'cancel',
            onPress: () => {
              this.props.navigation.goBack();
            },
          },
        ],
      );
    } finally {
      api.baseUrl = oldBaseUrl;
    }
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
  },
});

export default connect((state) => state.main)(AppsScreen);
