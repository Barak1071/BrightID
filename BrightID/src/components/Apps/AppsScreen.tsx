import React, {
  useCallback,
  useContext,
  useEffect,
  useState,
  useRef,
} from 'react';
import {
  Alert,
  StyleSheet,
  View,
  FlatList,
  Text,
  TextInput,
  StatusBar,
  RefreshControl,
  Animated,
  TouchableOpacity,
  Platform,
  SafeAreaView,
} from 'react-native';
import { useTranslation } from 'react-i18next';
import { any, propEq, find } from 'ramda';
import {
  useFocusEffect,
  useNavigation,
  useRoute,
} from '@react-navigation/native';
import { useHeaderHeight } from '@react-navigation/stack';
import _ from 'lodash';
import Spinner from 'react-native-spinkit';
import {
  linkedContextTotal,
  selectAllLinkedContexts,
  selectAllLinkedSigs,
} from '@/reducer/appsSlice';
import { useDispatch, useSelector } from '@/store';
import EmptyList from '@/components/Helpers/EmptyList';
import { ORANGE, BLUE, WHITE } from '@/theme/colors';
import { DEVICE_LARGE } from '@/utils/deviceConstants';
import {
  fetchApps,
  selectAllApps,
  selectPendingLinkedContext,
} from '@/actions';
import { fontSize } from '@/theme/fonts';
import { NodeApiContext } from '@/components/NodeApiGate';
import { isVerified } from '@/utils/verifications';
import AppCard from './AppCard';
import AnimatedLinearGradient from './AnimatedLinearGradient';
import { handleAppContext, handleBlindSigApp } from './model';

export const AppsScreen = () => {
  const filters = [
    { name: 'All Apps', id: 1 },
    { name: 'Linked', id: 2 },
    { name: 'Verified', id: 3 },
  ];

  const dispatch = useDispatch();
  const navigation = useNavigation();
  const route = useRoute<AppsRoute>();
  const api = useContext(NodeApiContext);
  const headerHeight = useHeaderHeight();
  const { t } = useTranslation();

  const apps = useSelector(selectAllApps);
  const linkedContext = useSelector(selectAllLinkedContexts);
  const linkedContextsCount = useSelector(linkedContextTotal);
  const selectLinkedSigs = useSelector(selectAllLinkedSigs);
  const isSponsored = useSelector((state: State) => state.user.isSponsored);
  const pendingLink = useSelector(selectPendingLinkedContext);
  const userVerifications = useSelector(
    (state: State) => state.user.verifications,
  );

  const [refreshing, setRefreshing] = useState(false);
  const [sponsoringApp, setSponsoringApp] = useState<AppInfo | undefined>(
    undefined,
  );
  const [totalApps, setTotalApps] = useState(0);
  const [totalVerifiedApp, setTotalVerifiedApp] = useState(0);

  // filter state
  const [search, setSearch] = useState('');
  const [activefilter, SetFilter] = useState(0);
  const [filteredApp, setFilteredApp] = useState(apps);

  // Animation
  const scrollY = useRef(new Animated.Value(0)).current;
  const fadeBackgroundSearch = scrollY.interpolate({
    inputRange: [0, 230],
    outputRange: [0, 1],
  });
  const fadeBackgroundHeader = scrollY.interpolate({
    inputRange: [0, 100],
    outputRange: [1, 0],
  });
  const handleScroll = Animated.event(
    [
      {
        nativeEvent: {
          contentOffset: { y: scrollY },
        },
      },
    ],
    {
      useNativeDriver: true,
    },
  );

  const refreshApps = useCallback(() => {
    setRefreshing(true);
    dispatch(fetchApps(api))
      .then(() => {
        setRefreshing(false);
      })
      .catch((err) => {
        console.log(err.message);
        setRefreshing(false);
      });
  }, [api, dispatch]);

  const handleDeepLink = useCallback(() => {
    const context = route.params?.context;
    const isValidContext = any(propEq('context', context))(apps);
    if (isValidContext) {
      handleAppContext(route.params);
    } else {
      Alert.alert(
        t('apps.alert.title.invalidContext'),
        t('apps.alert.text.invalidContext', { context: `${context}` }),
      );
    }
    // reset params
    navigation.setParams({
      baseUrl: '',
      context: '',
      contextId: '',
    });
  }, [navigation, route.params, apps, t]);

  const handleAppDeepLink = useCallback(() => {
    const appId = route.params?.context;
    const appInfo = find(propEq('id', appId))(apps) as AppInfo;
    if (api && appInfo && appInfo.usingBlindSig) {
      handleBlindSigApp(route.params, setSponsoringApp, api);
    } else {
      Alert.alert(
        t('apps.alert.title.invalidApp'),
        t('apps.alert.text.invalidApp', { app: `${appId}` }),
      );
    }
    // reset params
    navigation.setParams({
      context: '',
      contextId: '',
    });
  }, [route.params, apps, navigation, api, t]);

  useEffect(() => {
    if (apps.length > 0 && route.params?.baseUrl) {
      handleDeepLink();
    } else if (apps.length > 0 && route.params?.context) {
      handleAppDeepLink();
    }
  }, [apps, handleDeepLink, handleAppDeepLink, route.params]);

  // handle filter
  useEffect(() => {
    const allApps = apps.filter((app) => {
      const isLinked = linkedContext.find(
        (link) => link.context === app.context,
      );
      const isLinkedTemp = isLinked && isLinked.state === 'applied';
      return !app.testing || isLinkedTemp;
    });

    const linkedApps = allApps.filter((app) => {
      const appSig = selectLinkedSigs.filter((sig) => sig.app === app.id);
      const isLinked = linkedContext.find(
        (link) => link.context === app.context,
      );
      const isLinkedTemp = isLinked && isLinked.state === 'applied';
      return (app.usingBlindSig && appSig.length > 0) || isLinkedTemp;
    });

    const verifiedApps = allApps.filter((app) => {
      /* eslint no-var: 0 */
      var count = 0;
      for (const verification of app.verifications) {
        const verified = isVerified(
          _.keyBy(userVerifications, (v) => v.name),
          verification,
        );
        if (verified) {
          count++;
        }
      }

      return count === app.verifications.length;
    });

    setTotalApps(allApps.length);
    setTotalVerifiedApp(verifiedApps.length);

    var filteredApp: AppInfo[] = [];
    if (activefilter === 0) {
      filteredApp = allApps;
    } else if (activefilter === 1) {
      filteredApp = linkedApps;
    } else if (activefilter === 2) {
      filteredApp = verifiedApps;
    }

    // filter using search bar
    if (search !== '') {
      const filterResult = filteredApp.filter(
        (app) => app.name.toLowerCase().indexOf(search.toLowerCase()) !== -1,
      );
      setFilteredApp(filterResult);
    } else {
      setFilteredApp(filteredApp);
    }
  }, [
    search,
    activefilter,
    apps,
    selectLinkedSigs,
    linkedContext,
    userVerifications,
  ]);

  useFocusEffect(refreshApps);

  const AppStatus = () => {
    let msg: string, waiting: boolean;
    if (sponsoringApp) {
      msg = t('apps.text.sponsoring', { app: `${sponsoringApp.name}` });
      waiting = true;
    } else if (pendingLink) {
      msg = t('apps.text.pendingLink', { context: `${pendingLink.context}` });
      waiting = true;
    } else if (!isSponsored) {
      msg = t('apps.text.notSponsored');
      waiting = false;
    } else {
      msg = '';
      waiting = false;
    }
    return msg ? (
      <View style={styles.statusContainer}>
        <Text style={styles.statusMessage}>{msg}</Text>
        <Spinner
          isVisible={waiting}
          size={DEVICE_LARGE ? 48 : 42}
          type="Wave"
          color={BLUE}
        />
      </View>
    ) : (
      <View style={{ height: DEVICE_LARGE ? 12 : 10 }} />
    );
  };

  return (
    <>
      <View style={styles.container} testID="appsScreen">
        {/* <AppStatus /> */}

        <AnimatedLinearGradient
          containerStyle={{
            position: 'absolute',
            top: 0,
            width: '100%',
            height: headerHeight,
            zIndex: 100,
            opacity: fadeBackgroundSearch,
          }}
          colors={['#3E4481', '#999ECD']}
        />

        <AnimatedLinearGradient
          containerStyle={[
            styles.headerContainer,
            {
              opacity: fadeBackgroundHeader,
              transform: [
                {
                  translateY: scrollY.interpolate({
                    inputRange: [0, 150],
                    outputRange: [0, -150],
                    extrapolate: 'clamp',
                  }),
                },
              ],
            },
          ]}
          style={{ paddingHorizontal: 20, paddingTop: headerHeight }}
          colors={['#3E4481', '#999ECD', '#ED7A5D']}
        >
          <View style={styles.rowContainer}>
            <View style={styles.appDetailContainer}>
              <Text style={styles.detailLabel}>Total</Text>
              <Text style={styles.detail}>
                {totalApps} <Text style={styles.detailLabel}>apps</Text>
              </Text>
            </View>
            <View style={{ width: 10 }} />
            <View style={styles.appDetailContainer}>
              <Text style={styles.detailLabel}>You're linked to</Text>
              <Text style={styles.detail}>
                {linkedContextsCount}
                <Text style={styles.detailLabel}> apps</Text>
              </Text>
            </View>
          </View>
          <View style={styles.criteriaContainer}>
            <Text style={styles.detailLabel}>
              You meet the verification criteria of
            </Text>
            <Text style={[styles.detail, { textAlign: 'right' }]}>
              {totalVerifiedApp} <Text style={styles.detailLabel}>apps</Text>
            </Text>
          </View>
        </AnimatedLinearGradient>

        <Animated.View
          testID="searchBarBackground"
          style={{
            position: 'absolute',
            top: headerHeight,
            width: '100%',
            height: 120,
            backgroundColor: `white`,
            opacity: fadeBackgroundSearch,
            zIndex: 5,
          }}
        />

        <Animated.View
          style={{
            position: 'absolute',
            left: 0,
            right: 0,
            top: headerHeight + 10,
            alignSelf: 'center',
            alignItems: 'center',
            width: '100%',
            height: 100,
            zIndex: 6,
            overflow: 'visible',
            transform: [
              {
                translateY: scrollY.interpolate({
                  inputRange: [-500, 0, 150],
                  outputRange: [650, 150, 0],
                  extrapolate: 'clamp',
                }),
              },
            ],
          }}
        >
          <TextInput
            style={[styles.shadow, styles.textInput]}
            onChangeText={(value) => setSearch(value)}
            placeholder="App name"
          />

          <View style={styles.filterContainer}>
            {filters.map((item, index) => (
              <TouchableOpacity
                key={index}
                style={[
                  styles.filterItemContainer,
                  { backgroundColor: index === activefilter ? ORANGE : WHITE },
                ]}
                onPress={() => SetFilter(index)}
              >
                <Text
                  style={{
                    fontFamily: 'Poppins-Medium',
                    fontSize: 12,
                    color: index === activefilter ? WHITE : ORANGE,
                  }}
                >
                  {item.name}
                </Text>
              </TouchableOpacity>
            ))}
          </View>
        </Animated.View>

        <Animated.FlatList
          testID="appsList"
          data={filteredApp}
          onScroll={handleScroll}
          scrollEventThrottle={16}
          numColumns={2}
          columnWrapperStyle={styles.wrapperColumn}
          contentContainerStyle={[
            styles.contentContainer,
            { marginTop: headerHeight + 190 },
          ]}
          keyExtractor={({ name }, index) => name + index}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          renderItem={({ item }) => <AppCard {...item} />}
          ListEmptyComponent={
            <EmptyList title={t('apps.text.noApps')} iconType="flask" />
          }
          refreshControl={
            <RefreshControl refreshing={refreshing} onRefresh={refreshApps} />
          }
        />
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  orangeTop: {
    backgroundColor: ORANGE,
    height: DEVICE_LARGE ? 70 : 65,
    width: '100%',
    zIndex: 1,
  },
  container: {
    flex: 1,
  },
  shadow: {
    ...Platform.select({
      android: { shadowColor: 'rgba(0,0,0,1)' },
      ios: { shadowColor: 'rgba(0,0,0,0.2)' },
    }),
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 1,
    shadowRadius: 4,
    elevation: 5,
  },
  centerItem: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  statusContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 20,
    paddingBottom: 20,
  },
  statusMessage: {
    fontFamily: 'Poppins-Medium',
    textAlign: 'center',
    fontSize: fontSize[14],
    color: BLUE,
  },
  linkingContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: 20,
  },
  headerContainer: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    height: 310,
    width: '100%',
  },
  rowContainer: {
    flexGrow: 0.4,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  appDetailContainer: {
    flex: 1,
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
    padding: 12,
    borderRadius: 10,
  },
  criteriaContainer: {
    flexDirection: 'row',
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
    padding: 12,
    borderRadius: 10,
    justifyContent: 'space-between',
    marginTop: 10,
    alignItems: 'center',
  },
  detailLabel: {
    flex: 1,
    fontFamily: 'Poppins-Light',
    fontSize: 13,
    color: WHITE,
  },
  detail: {
    flex: 1,
    fontFamily: 'Poppins-Bold',
    fontSize: 20,
    color: WHITE,
  },
  textInput: {
    backgroundColor: 'white',
    width: '90%',
    height: 50,
    borderRadius: 5,
    paddingHorizontal: 10,
  },
  wrapperColumn: {
    justifyContent: 'space-between',
    marginBottom: 20,
  },
  contentContainer: {
    paddingTop: 100,
    paddingBottom: 500,
    paddingHorizontal: 20,
    backgroundColor: 'white',
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
  },
  filterContainer: {
    width: '90%',
    flexDirection: 'row',
    marginTop: 10,
  },
  filterItemContainer: {
    borderWidth: 1,
    borderColor: ORANGE,
    padding: 5,
    borderRadius: 5,
    marginRight: 5,
  },
});

export default AppsScreen;
