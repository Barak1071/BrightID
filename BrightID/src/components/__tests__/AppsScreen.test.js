import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { AppsScreen } from '../Apps/AppsScreen';

// you can mock any middlewares here if necessary

const props = {
  navigation: { navigate: jest.fn() },
};

describe('Testing AppsScreen', () => {
  it('renders as expected', () => {
    const wrapper = renderer.create(<AppsScreen {...props} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
