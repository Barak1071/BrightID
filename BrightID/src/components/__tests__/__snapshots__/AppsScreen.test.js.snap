// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AppsScreen renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fdfdfd",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopColor": "rgba(0, 0, 0, .3)",
        "borderTopWidth": 0.5,
        "flexDirection": "row",
        "height": 63,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityLabel="Home"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Home
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Connections"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Connections
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Groups"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Groups
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Apps"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Notifications
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Apps"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Apps
        </Text>
      </View>
    </View>
  </View>
</View>
`;
